lib/src/time-range-dialog.dart

```diff
@@ -637,7 +640,11 @@ class _TimeRangePickerState extends State<TimeRangePicker>
             children: [
               Text(widget.fromText, style: TextStyle(color: activeColor)),
               Text(
-                _startTime != null ? _startTime.format(context) : "-",
+                _startTime != null
+                    ? MaterialLocalizations.of(context).formatTimeOfDay(
+                        _startTime,
+                        alwaysUse24HourFormat: widget.use24HourFormat)
```

Comment:
this was pretty much it, we just needed to pass this to the localization function.